Stripe/StripeiOSTests/PaymentSheetPaymentMethodTypeTest.swift

```diff
@@ -298,6 +298,26 @@ class PaymentSheetPaymentMethodTypeTest: XCTestCase {
         XCTAssertEqual(types[0], .card)
     }
 
+    func testPaymentIntentFilteredPaymentMethodTypes_withSetupFutureUsage() {
+        let paymentIntent = constructPI(
+            paymentMethodTypes: ["card", "klarna", "p24"],
+            orderedPaymentMethodTypes: ["card", "klarna", "p24"],
+            setupFutureUsage: .onSession
+        )!
+        let intent = Intent.paymentIntent(paymentIntent)
+        var configuration = PaymentSheet.Configuration()
+        configuration.returnURL = "http://return-to-url"
+        configuration.allowsDelayedPaymentMethods = true
+        let types = PaymentSheet.PaymentMethodType.filteredPaymentMethodTypes(
+            from: intent,
+            configuration: configuration
+        )
+
+        XCTAssertEqual(types.count, 1)
+        XCTAssertEqual(types[0], .card)
+        // Klarna and P24 do not support save and reuse so they should be filtered out
```

Comment:
Should you add cash app to this test and `testSetupIntentFilteredPaymentMethodTypes`?